FlowCrypt/Functionality/Mail Provider/Message Provider/MessageService.swift

```diff
@@ -278,35 +277,28 @@ final class MessageService {
 
 // MARK: - Message verification
 extension MessageService {
-    private func fetchVerificationPubKeys(for sender: String?) -> [String] {
-        if let sender = sender {
-            return contactsService.retrievePubKeys(for: sender)
-        } else {
-            return []
-        }
+    private func fetchVerificationPubKeys(for sender: String?) async throws -> [String] {
+        guard let sender = sender else { return [] }
+
+        let pubKeys = contactsService.retrievePubKeys(for: sender)
+        if pubKeys.isNotEmpty { return pubKeys }
+
+        guard let contact = try? await contactsService.searchContact(with: sender)
+        else { return [] }
+
+        return contact.activePubKeys.map(\.armored)
     }
 
     private func evaluateSignatureVerificationResult(
-        signature: MsgBlock.VerifyRes?,
-        sender: String?
+        signature: MsgBlock.VerifyRes?
     ) async -> ProcessedMessage.MessageSignature {
         guard let signature = signature else { return .unsigned }
 
-        if let error = signature.error {
-            return .error(error)
-        }
+        if let error = signature.error { return .error(error) }
 
         guard let signer = signature.signer else { return .unsigned }
 
-        var pubKey: PubKey?
-
-        if let contact = await contactsService.findBy(longId: signer) {
-            pubKey = contact.pubKey(with: signer)
-        } else if let email = sender, let contact = try? await contactsService.searchContact(with: email) {
-            pubKey = contact.pubKey(with: signer)
-        }
-
-        guard pubKey != nil && signature.match != nil else { return .missingPubkey(signer) }
+        guard signature.match != nil else { return .missingPubkey(signer) }
 
         guard signature.match == true else { return .bad }
 
```

Comment:
Just realized - there is more than one kind of `.good` :-) TypeScript code for reference:

```ts
  if (verifyRes && verifyRes.match) {
    if (mixedSignatures) {
      verifyRes.mixed = true;
    }
    if (signedBlockCount > 0 && signedBlockCount != contentBlocks.length) {
      verifyRes.partial = true;
    }
  }
```

That means `VerifyRes` needs to be updated to also include `mixed: Bool?` and `partial: Bool?`. Plus this logic here, too. But that's for another issue, needs more thought. It won't be a difficult change once thought through.